### PR TITLE
Adding @Deprecated annotation when API is marked as deprecated

### PR DESCRIPTION
**Pull request by shorea**
This change generates the @Deprecated annotation for all APIs marked as deprecated in the C2J model.

## Motivation and Context

This adds clear guidance for deprecated APIs in the IDE.

N/A

## Modifications

Modifies the code generator to add the @Deprecated annotation to client/interface methods that correspond to a deprecated API.

## Testing

Added deprecated trait to one of the existing poet test models and modified expected generation files.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)

## Checklist

- [ x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x ] Local run of `mvn install` succeeds
- [x ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x ] I have added tests to cover my changes
- [x ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Review by zoewangg — APPROVED** [review]
Hey Andrew! 👋🏽 :shipit:

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java [hunk 1]
```diff
@@ -160,6 +160,7 @@ public final String serviceName() {
      * @see <a href="https://docs.aws.amazon.com/goto/WebAPI/json-service-2010-05-08/APostOperation" target="_top">AWS
      *      API Documentation</a>
      */
+    @Deprecated
```
Comment (zoewangg) [inline codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java:163]: Looks like there's a `deprecatedMessage` field. Can we add it in the Javadocs?

Comment (Bennett-Lynch) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java:163]: We can probably update this utility class to support operation models as well:

https://github.com/aws/aws-sdk-java-v2/blob/a8ec3f8c1aaf81e06394b30daec2074562581c70/codegen/src/main/java/software/amazon/awssdk/codegen/poet/model/DeprecationUtils.java#L34-L50

Comment (shorea) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java:163]: Oh nice @Bennett-Lynch this looks perfect, will make some tweaks to use that.

@zoewangg sure thing! :)

Comment (shorea) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java]: Added those changes. Let me know what you think!

**Review by Bennett-Lynch — APPROVED** [review]

File: codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java (hunk 1 above)
Comment (Bennett-Lynch) [reply codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-async-client-class.java]: LGTM!

**Review by zoewangg — APPROVED** [review]

**Pull request merged**